src/component/NewsFeed.js

```diff
@@ -3,15 +3,15 @@ import News from "../component/News";
 import styles from "../component/NewsFeed.module.css";
 export default class NewsFeed extends Component {
   componentWillMount() {
-    this.props.NewsData();
+    this.props.data();
   }
   render() {
     return (
       <div className={styles.base}>
-        {this.props.newsfeed &&
-          this.props.newsfeed.newsData &&
-          this.props.newsfeed.newsData &&
-          this.props.newsfeed.newsData.articles.map(news => (
+        {this.props.newsFeed &&
+          this.props.newsFeed.newsFeedData &&
+          this.props.newsFeed.newsFeedData &&
+          this.props.newsFeed.newsFeedData.articles.map(news => (
```

Comment:
add this.props.newsFeedData.articles&&
because if there is no articles node in the newsFeedData  the map() will trows error